packages/compiler/src/core/checker.ts

```diff
@@ -3001,6 +3065,43 @@ export function createChecker(program: Program): Checker {
     return type;
   }
 
+  function checkModelValidate(
+    vv: ModelValidateNode,
+    mapper: TypeMapper | undefined
+  ): ModelValidate {
+    const links = getSymbolLinksForMember(vv);
+    if (links && links.declaredType && mapper === undefined) {
+      return links.declaredType as ModelValidate;
+    }
+
+    const name = vv.id?.sv ?? "[Anonymous]";
+
+    //
+    //TODO: typecheck the projection expression here or leave that problem for the emitter?
```

Comment:
One thing I think we should check, at least: validate that the methods you invoke are present. We should have a standard built-in set of operations on types, it should be an error to say like `arr::lenght`. In the future we could discuss having some kind of extensibility mechanism here but I would like to start as locked-down as possible.